core/src/main/java/org/elasticsearch/script/ScriptParameterParser.java

```diff
@@ -59,7 +53,7 @@ public ScriptParameterParser(Set<String> parameterNames) {
             fileParameters = new HashSet<>();
             indexedParameters = new HashSet<>();
             for (String parameterName : parameterNames) {
-                if (ScriptService.SCRIPT_LANG.match(parameterName)) {
+                if (ScriptService.SCRIPT_LANG.match(parameterName, ParseField.EMPTY_FLAGS)) {
```

Comment:
Here we should be able to get correct flags? Maybe not always, but at least in the context of aggs, I think the search context is available